### PR TITLE
OPCM: AddressManager and ProxyAdmin assertions.

### PR DESCRIPTION
**Pull request by blmalone**
Adding additional assertions for:
- AddressManager
- ProxyAdmin
- and all Proxy implementations (3 types of proxy)

### CODE REVIEW

**Comment by blmalone** [comment]
* **#12370** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12370?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @blmalone and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/scripts/libraries/DeployUtils.sol [hunk 1]
```diff
@@ -217,11 +219,28 @@ library DeployUtils {
 
     /// @notice Asserts that the given proxy has an implementation set.
     /// @param _proxy Proxy to check.
-    function assertImplementationSet(address _proxy) internal {
+    function assertERC1967ImplementationSet(address _proxy) internal returns (address implementation) {
```
Comment (blmalone) [inline packages/contracts-bedrock/scripts/libraries/DeployUtils.sol]: Originally had this function generic e.g. `assertImplementationSet` but it got messy. I ended up having to pass through 4 params i.e. `addressManager, proxy, proxyType, implementationName`. 

Splitting them was the cleaner approach.

File: packages/contracts-bedrock/scripts/DeployOPChain.s.sol [hunk 2]
```diff
@@ -20,6 +20,7 @@ import { IProxyAdmin } from "src/universal/interfaces/IProxyAdmin.sol";
 import { IProxy } from "src/universal/interfaces/IProxy.sol";
 
 import { IAddressManager } from "src/legacy/interfaces/IAddressManager.sol";
+import { IL1ChugSplashProxy } from "src/legacy/interfaces/IL1ChugSplashProxy.sol";
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/DeployOPChain.s.sol]: unused import CI check is failing

Comment (blmalone) [reply packages/contracts-bedrock/scripts/DeployOPChain.s.sol]: fixed in: https://github.com/ethereum-optimism/optimism/pull/12370/commits/e6e644b412a530dc3df109db354fd7e12005346d

File: packages/contracts-bedrock/scripts/libraries/DeployUtils.sol [hunk 3]
```diff
@@ -217,12 +222,89 @@ library DeployUtils {
...
+    )
+        internal
+        view
+        returns (address implementation_)
+    {
+        implementation_ = _addressManager.getAddress(_implementationName);
+        assertValidContractAddress(implementation_);
+    }
+
+    function buildERC1967ProxyWithImpl(string memory proxyImplName) public returns (IProxy genericProxy) {
```
Comment (blmalone) [inline packages/contracts-bedrock/scripts/libraries/DeployUtils.sol]: Helper functions for building the three types of proxies with a dummy implementation (i.e. code at impl addr is `0x01`).

**Review by mds1 — APPROVED** [review]

**Pull request merged**